### DISCUSSION TITLE
Any updates on the time picker component?

### DISCUSSION THREAD

**Discussion by Thesmit113** [chakra-ui/ark#3020] [General]
In the changelog, I noticed the following entry :
[4.1.1] - 2024-10-09
Changed
TimePicker [Preview]: Updated value and defaultValue types from string to Time. Use the exported parseTime function to convert between strings and time objects.

However, I couldn't find any corresponding updates in docs

**Comment by cschroeter** [comment]
@Thesmit113 

Hey mate, the `TimePicker` has already been merged and you play around if you are adventurous. Here are the stories we have at the moment:

https://github.com/chakra-ui/ark/tree/main/packages/react/src/components/time-picker/examples

**Reply by Tihomir971** [reply]
TimePicker is no longer planned?

**Comment by segunadebayo** [comment]
We've decided not to ship a dedicated TimePicker component. Time selection UIs vary widely across products, so we think this is better handled in userland.

Here are two patterns you can build with existing Ark components:

**1. Time Select (using Select)**
```tsx
import { Select, createListCollection } from '@ark-ui/react/select'

const times = createListCollection({
  items: Array.from({ length: 24 * 4 }, (_, i) => {
    const h = String(Math.floor(i / 4)).padStart(2, '0')
    const m = String((i % 4) * 15).padStart(2, '0')
    return { label: `${h}:${m}`, value: `${h}:${m}` }
  }),
})

<Select.Root collection={times}>
  <Select.Label>Time</Select.Label>
  <Select.Control>
    <Select.Trigger>
      <Select.ValueText placeholder="Select time" />
    </Select.Trigger>
  </Select.Control>
  <Select.Positioner>
    <Select.Content>
      {times.items.map((item) => (
        <Select.Item key={item.value} item={item}>
          <Select.ItemText>{item.label}</Select.ItemText>
        </Select.Item>
      ))}
    </Select.Content>
  </Select.Positioner>
</Select.Root>
```

**2. Time Grid**
```tsx
const TimeGrid = ({ onSelect }) => {
  const hours = Array.from({ length: 24 }, (_, i) => String(i).padStart(2, '0'))
  const minutes = ['00', '15', '30', '45']

  return (
    <div style={{ display: 'grid', gridTemplateColumns: 'repeat(4, 1fr)', gap: '4px' }}>
      {hours.flatMap((h) =>
        minutes.map((m) => (
          <button key={`${h}:${m}`} onClick={() => onSelect(`${h}:${m}`)}>
            {h}:{m}
          </button>
        )),
      )}
    </div>
  )
}
```
